front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy

```diff
@@ -117,13 +117,15 @@ public class ApplicationsController {
   @RequestMapping(method = RequestMethod.GET, value = "/{applicationName:.+}")
   Application get(@PathVariable final String applicationName) {
     def app = applicationDAO.findByName(applicationName.toUpperCase())
-    try {
-      def perm = applicationPermissionDAO?.findById(app.name)
-      if (perm?.permissions?.isRestricted()) {
-        app.details().put("permissions", perm.permissions)
+    if (app != null) {
```

Comment:
do you know what causes the app not to be found in applicationDAO?

Reply:
not sure yet. while after the fixing, create application succeeded. Isn't the get before creating, so not found?

Reply:
You need to dig a bit deeper and understand how applicationDAO is built and if the answer it might be that is not fully supported for Azure. It looks like it might have to do with some permission setting so may be we are missing something during the Front50 configuration? 
And since this change is in common code you will have to make a compelling case when pushing the PR into main branch why adding this check is needed.

Reply:
that make sense. Actually I'm looking at permission check support for azure part. For the null check itself, no matter which provider it is, if app is null, later call on app.name will fail with null object error.

Reply:
A shorter way to express this is in groovy: 
```
if (app && app.name)
```

Reply:
thanks you @ttomsu for the comments. while besides directly fix this null error, we'd like to learn more about permission feature in spinnaker. what behavior should be added in cloud provider to support permission when creating application?